src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultProjectSnapshotManager.cs

```diff
@@ -193,7 +193,7 @@ public override void DocumentAdded(HostProject hostProject, HostDocument documen
             {
                 var loader = textLoader is null
                     ? DocumentState.EmptyLoader
-                    : (() => textLoader.LoadTextAndVersionAsync(Workspace, documentId: null, CancellationToken.None));
+                    : (() => textLoader.LoadTextAndVersionAsync(default, CancellationToken.None));
```

Comment:
I don't remember the specifics because it's been a bit, but IIRC if this doesn't set `SourceHashAlgorithm` to something other than `None` you're gonna run into failures from this.

Reply:
Those failures were hidden for us, so I'd recommend debugging through this to verify that it doesn't throw.